src/vmm/src/vstate/memory.rs

```diff
@@ -55,37 +55,37 @@ pub enum MemoryError {
     HugetlbfsSnapshot,
 }
 
+/// Enum representing the type of memory backing the virtual machine
+#[derive(Debug)]
+pub enum MemoryKind {
+    /// MAP_PRIVATE | MAP_ANONYMOUS should be used
+    MapAnonymous,
+    /// A memfd mapped as MAP_SHARED should be used (in case guest memory needs to be
+    /// shared with other processes)
+    Memfd,
+    /// The given snapshot file should be mapped into the geust
+    Snapshot(File),
```

Comment:
It looks like we'd like to map the "purpose" of the memory backend (ie booting/restoring x sendable/unsendable x hp/non-hp) to physical mmap flags. I'm wondering of a way to make this mapping more explicit somehow as currently "Snapshot" means "Snapshot/file", but not "Snapshot/uffd", which is slightly misleading to me.

Reply:
I've shared this to `FilePrivate` based on our offline discussion.